src/plugins/hidsim/gui/vmi_win_gui_utils.h

```diff
@@ -0,0 +1,220 @@
+ * If you have received a written license agreement or contract for        *
+ * Covered Software stating terms other than these, you may choose to use  *
+ * and redistribute Covered Software under those terms instead of these.   *
+ *                                                                         *
+ * Source is provided to this software because we believe users have a     *
+ * right to know exactly what a program is going to do before they run it. *
+ * This also allows you to audit the software for security holes.          *
+ *                                                                         *
+ * Source code also allows you to port DRAKVUF to new platforms, fix bugs, *
+ * and add new features.  You are highly encouraged to submit your changes *
+ * on https://github.com/tklengyel/drakvuf, or by other methods.           *
+ * By sending these changes, it is understood (unless you specify          *
+ * otherwise) that you are offering unlimited, non-exclusive right to      *
+ * reuse, modify, and relicense the code.  DRAKVUF will always be          *
+ * available Open Source, but this is important because the inability to   *
+ * relicense code has caused devastating problems for other Free Software  *
+ * projects (such as KDE and NASM).                                        *
+ * To specify special license conditions of your contributions, just say   *
+ * so when you send them.                                                  *
+ *                                                                         *
+ * This program is distributed in the hope that it will be useful, but     *
+ * WITHOUT ANY WARRANTY; without even the implied warranty of              *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the DRAKVUF   *
+ * license file for more details (it's in a COPYING file included with     *
+ * DRAKVUF, and also available from                                        *
+ * https://github.com/tklengyel/drakvuf/COPYING)                           *
+ *                                                                         *
+ * This file was created by Jan Gruber.                                    *
+ * It is distributed as part of DRAKVUF under the same license             *
+ ***************************************************************************/
+
+#ifndef VMI_WIN_GUI_UTILS_H
+#define VMI_WIN_GUI_UTILS_H
+
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <inttypes.h>
+#include <wchar.h>
+#include <getopt.h>
+
+#include <libvmi/libvmi.h>
+
+#include "vmi_win_gui_offsets.h"
+
+extern struct Offsets symbol_offsets;
+
+#define ARRAY_SIZE(x) (sizeof(x) / sizeof((x)[0]))
+
+/*
+ * Checks, whether the _OBJECT_HEADER, which precedes every executive object,
+ * is preceded by an optional header of type _OBJECT_HEADER_NAME_INFO.
+ * If this is the case, the name of the executive object is read and returned.
+ *
+ */
+char* retrieve_objhdr_name(vmi_instance_t vmi, addr_t addr)
```

Comment:
why function definitions are placed in header ".h" files?

Reply:
I thought, it is okay to place it in there, since some plugins also house function implementations in their `private.h` (e.g. https://github.com/tklengyel/drakvuf/blob/master/src/plugins/tlsmon/private.h) and the overhead is negligible. 

Should I separate all implementations from their declarations in the .h-files in `./src/plugins/hidsim/gui/`?

Reply:
I'd like implementations in separate file - it's easier for someone who wants to understand the code to just read .h with comments and signatures instead of scrolling 100LoC of function body.

Reply:
Changed in 512cd01a19b8ee1822ea4fc9f3f00ca309c655fc